File: Output/javascript/hard/SubsequenceWithTheMinimumScore.js
/**https://leetcode.com/problems/subsequence-with-the-minimum-score/ */
//You are given two strings s and t.
//You are allowed to remove any number of characters from the string t.
//The score of the string is 0 if no characters are removed from the string t, otherwise:
//Let left be the minimum index among all removed characters.
//Let right be the maximum index among all removed characters.
//Then the score of the string is right - left + 1.
//Return the minimum possible score to make t a subsequence of s.
//A subsequence of a string is a new string that is formed from the original string by deleting some (can be none) of the characters without disturbing the relative positions of the remaining characters. (i.e., "ace" is a subsequence of "abcde" while "aec" is not).


/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var minimumScore = function(s, t) {
    let n = s.length;
    let m = t.length;
    let dp = [];
    for(let i = 0; i <= n; i++){
        dp.push([]);
        for(let j = 0; j <= m; j++){
            dp[i].push(0);
        }
    }
    for(let i = 1; i <= n; i++){
        for(let j = 1; j <= m; j++){
            if(s[i - 1] === t[j - 1]){
                dp[i][j] = dp[i - 1][j - 1] + 1;
            }else{
                dp[i][j] = Math.max(dp[i - 1][j], dp[i][j - 1]);
            }
        }
    }
    let result = 0;
    for(let i = 1; i <= n; i++){
        for(let j = 1; j <= m; j++){
            if(dp[i][j] === dp[n][m]){
                result = Math.max(result, i - j + 1);
            }
        }
    }
    return result;
    
};

    

File: Tool/PreUtil/src/output/c/hard/SubsequenceWithTheMinimumScore.c
/**https://leetcode.com/problems/subsequence-with-the-minimum-score/ */
//You are given two strings s and t.
//You are allowed to remove any number of characters from the string t.
//The score of the string is 0 if no characters are removed from the string t, otherwise:
//Let left be the minimum index among all removed characters.
//Let right be the maximum index among all removed characters.
//Then the score of the string is right - left + 1.
//Return the minimum possible score to make t a subsequence of s.
//A subsequence of a string is a new string that is formed from the original string by deleting some (can be none) of the characters without disturbing the relative positions of the remaining characters. (i.e., "ace" is a subsequence of "abcde" while "aec" is not).


int minimumScore(char * s, char * t){
    int i, j, k, l, m, n, o, p, q, r, a, b, c, d, e, f, g, h, w, x, y, z;
    int *s = (int *)malloc(sizeof(int) * (n + 1));
    int *t = (int *)malloc(sizeof(int) * (n + 1));
    int *a = (int *)malloc(sizeof(int) * n);
    for (i = 0; i < n; i++) {
        a[i] = i + 1;
    }
    for (i = 0; i < n; i++) {
        f = 1;
        for (j = 1; j <= n - i - 1; j++) {
            f *= j;
        }
        t = (k - 1) / f;
        s[i] = a[t] + '0';
        for (j = t; j < n - i - 1; j++) {
            a[j] = a[j + 1];
        }
        k -= t * f;
    }
    s[n] = '\0';
    return s;

}

    